go-controller/pkg/node/linkmanager/link_network_manager.go

```diff
@@ -280,3 +282,29 @@ func containsAddress(addresses []netlink.Addr, candidate netlink.Addr) bool {
 	}
 	return false
 }
+
+func gratuitousArpOverIfaceByName(srcIP net.IP, ifaceName string) error {
+	iface, err := net.InterfaceByName(ifaceName)
+	if err != nil {
+		return fmt.Errorf("failed finding interface '%s': %w", ifaceName, err)
+	}
+
+	c, err := arp.Dial(iface)
+	if err != nil {
+		return fmt.Errorf("failed dialing logical switch interface '%s': %w", ifaceName, err)
+	}
+	defer c.Close()
+	addr, err := netip.ParseAddr(srcIP.String())
+	if err != nil {
+		return fmt.Errorf("failed converting net.IP to netip.Addr: %w", err)
+	}
+	p, err := arp.NewPacket(arp.OperationRequest, iface.HardwareAddr, addr, net.HardwareAddr{0, 0, 0, 0, 0, 0}, addr)
```

Comment:
any particular reason for using 0's for `dstHW` ? can we use broadcasting mac ?

Reply:
Took an example from here 
https://github.com/higebu/gogarp/blob/master/main.go#L40

The real dst is used at `WriteTo` function later on.